feat: configurable timestamp format

ydylla opened the pull request:
Hi @problame,
this is a continuation of PR #466 where you proposed a new config option to change the timestamp format that is used in the snapshot names.  
The new option is called `utc_timestamp_format` and can be used with the `cron` and `periodic` job types. It defaults to the old timestamp format (which is now called `dense`) so existing installations should not break.  
People that want to change it can now choose between `dense`, `human`, `iso-8601`, `unix-seconds` or any custom format that is accepted by Go's `time.Time#Format`.

I also tried to add docs but I am not sure how they look rendered, since I did not want to setup sphinx. Maybe you can take a look and ensure they look nice?

Closes #465

problame reviewed (changes requested):
Thanks for this contribution, looks high quality to me, and we can merge it for 0.6.
I only have one nit that needs to be addressed before merging.

Also, if it's not too much to ask, I'd appreciate if you added `config_snapshotting_test*` cases for all 4 variants. You can copy-paste `TestSnapshotting` and do the same thing we do there, but for the `utc_timestamp_format`.

Lastly, I won't make a thing out of it, but I think `timestamp_format` is sufficient, don't you think? It's an optional flag, so, anyone who is interested in changing the default will probably already know that we use UTC there.

ydylla commented:
> Thanks for this contribution, looks high quality to me, and we can merge it for 0.6.

Awesome, thanks.

> but I think timestamp_format is sufficient

Yes I agree, I was just using the name you mentioned in #466.  
I have renamed the option and also added the default tags.

> I'd appreciate if you added config_snapshotting_test* cases for all 4 variants

I do not see any value in doing so, since the test only covers the config parsing. I would trust the yaml lib that it can handle different values 😄 But you are right there is no test that covers the `formatNow` function where the configured format is converted into an actual timestamp format. With the current code I do not see an easy way to write one.

ydylla commented on daemon/snapper/impl.go:
```diff
@@ -58,6 +60,21 @@ type snapProgress struct {
+	now := time.Now().UTC()
+	switch strings.ToLower(format) {
+	case "", "dense":
+		format = "20060102_150405_000"
```
I added the default tags and removed the `""` case.

problame commented:
> I do not see any value in doing so, since the test only covers the config parsing. I would trust the yaml lib that it can handle different values smile But you are right there is no test that covers the formatNow function where the configured format is converted into an actual timestamp format. With the current code I do not see an easy way to write one.

Yeah, the situation kind of sucks / it's repetitive code. I like to test full .yml with all the different config options because it serves as a poor regression test for inadvertent config format changes.

But as I said, we can go without that.

Thanks again for your contribution, I'll pull this in before the next release, which will happen this month.

ydylla commented:
> I like to test full .yml with all the different config options because it serves as a poor regression test for inadvertent config format changes

Ah I understand. I have added a `cron` case to the `TestSnapshotting` test and added a test that ensures the timestamp_format defaults to `dense`.
